fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -0,0 +1,166 @@
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Top-level UI for displaying the navigation bar.
+ *
+ * @param actionItems A list of [ActionItem] used to populate the bar.
+ */
+@Composable
+fun NavigationBar(actionItems: List<ActionItem>) {
+    Box(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .height(48.dp)
+            .fillMaxWidth(),
+    ) {
+        Divider(color = FirefoxTheme.colors.borderPrimary)
+
+        Row(
+            modifier = Modifier
+                .align(Alignment.Center)
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp),
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            actionItems.forEach {
+                when (it.type) {
+                    ItemType.STANDARD -> {
+                        IconButton(onClick = {}) {
+                            Icon(
+                                painter = painterResource(it.iconId),
+                                stringResource(id = it.descriptionResourceId),
+                                tint = FirefoxTheme.colors.iconPrimary,
+                            )
+                        }
+                    }
+                    ItemType.TAB_COUNTER -> {
+                        IconButton(onClick = {}) {
+                            TabsIcon(item = it, tabsCount = 0)
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun TabsIcon(item: ActionItem, tabsCount: Int) {
+    Box(contentAlignment = Alignment.Center) {
+        Icon(
+            painter = painterResource(id = item.iconId),
+            contentDescription = stringResource(id = item.descriptionResourceId),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+
+        Text(
+            text = tabsCount.toString(),
+            color = FirefoxTheme.colors.iconPrimary,
+            fontSize = 10.sp,
+            fontWeight = FontWeight.W700,
+            textAlign = TextAlign.Center,
+        )
+    }
+}
+
+/**
+ * Represents a navigation bar element.
+ *
+ * @property iconId Resource ID of the icon that item should display.
+ * @property descriptionResourceId Text used as a content description by accessibility services.
+ * @property type Type of the item, defaults to [ItemType.STANDARD].
+ */
+data class ActionItem(
+    val iconId: Int,
+    val descriptionResourceId: Int,
+    val type: ItemType = ItemType.STANDARD,
+)
+
+/**
+ * Enumerates the types of items that can be used in a navigation bar.
+ *
+ * [STANDARD] - Represents a regular navigation item. Used for most navigation actions.
+ * [TAB_COUNTER] - Represents a specialized item used to display a count, such as the number of open tabs in a browser.
+ */
+enum class ItemType {
+    STANDARD, TAB_COUNTER
+}
+
+/**
+ * Provides a collection of standard navigation items used in the application's navigation bar.
+ */
+object StandardNavigationItems {
```

Comment:
```suggestion
object NavigationItems {
```
Perhaps a simpler name here unless there's a different use case where the navbar becomes non-standard.